src/Build/BackEnd/Components/BuildRequestEngine/BuildRequestEngine.cs

```diff
@@ -331,7 +331,10 @@ public void SubmitBuildRequest(BuildRequest request)
                         // Grab the results from the requested configuration
                         IResultsCache cache = (IResultsCache)_componentHost.GetComponent(BuildComponentType.ResultsCache);
                         BuildResult result = cache.GetResultsForConfiguration(request.ConfigurationId);
-                        BuildResult resultToReport = new BuildResult(request, result, null);
+                        BuildResult resultToReport = new BuildResult(request, result, null)
+                        {
+                            BuildRequestDataFlags = request.BuildRequestDataFlags,
```

Comment:
This seems like something that should be set in the constructor, not manually here.